src/view/filter_mode/node_editor.py

```diff
@@ -0,0 +1,222 @@
+"""Widget to create multiple scenes and manage filters.
+
+Usage (where self is a QWidget and board_configuration is a BoardConfiguration):
+    node_editor = NodeEditor(self, board_configuration)
+    self.addWidget(node_editor)
+"""
+# coding=utf-8
+from PySide6 import QtGui, QtWidgets
+from PySide6.QtWidgets import QWidget, QTabWidget, QInputDialog
+from pyqtgraph.flowchart.NodeLibrary import NodeLibrary
+from pyqtgraph.flowchart import Flowchart
+
+from model.board_configuration import BoardConfiguration, Scene
+from file.read import read_document
+from file.write import create_xml, write_document
+from . import nodes
+from .scene_tab import SceneTabWidget
+
+class NodeEditorWidget(QTabWidget):
+    """Node Editor to create and manage filters."""
+
+    def __init__(self, parent: QWidget, board_configuration: BoardConfiguration) -> None:
+        super().__init__(parent)
+
+        self._library = NodeLibrary()
+
+        self._tab_widgets: list[SceneTabWidget] = []
+
+        self._board_configuration = board_configuration
+
+        self._register_constants_nodes()
+        self._register_debug_nodes()
+        self._register_adapters_nodes()
+        self._register_time_nodes()
+        self._register_fader_nodes()
+
+        # Buttons to add or remove scenes from show
+        self.addTab(QWidget(), "+")
+        self.addTab(QWidget(), "-")
+
+        self.tabBarClicked.connect(self._tab_bar_clicked)
+
+        # Toolbar for io/network actions
+        self._toolbar: list[QtGui.QAction] = []
+        save_show_file_button = QtGui.QAction("save Scene")
+        load_show_file_button = QtGui.QAction("load Scene")
+        enter_scene_button = QtGui.QAction("enter Scene")
+        send_show_button = QtGui.QAction("send Scene")
+
+        enter_scene_button.triggered.connect(self._enter_scene)
+        send_show_button.triggered.connect(self._send_show_file)
+        save_show_file_button.triggered.connect(lambda: self._select_file(self._save_show_file))
+        load_show_file_button.triggered.connect(lambda: self._select_file(self._load_show_file))
+
+        self._toolbar.append(enter_scene_button)
+        self._toolbar.append(send_show_button)
+        self._toolbar.append(save_show_file_button)
+        self._toolbar.append(load_show_file_button)
+
+    def _register_constants_nodes(self):
+        """Registers all the constants nodes."""
+        # Add Node -> Constants sub menu
+        self._library.addNodeType(nodes.Constants8BitNode, [('Constants',)])
```

Comment:
In order to improve code navigation capabilities, I suggest extracting these filter meta data definitions to a separate file and only iterate through them here. One should look out for a way to reduce the size of files larger than ~150 lines and this is a quite easy thing to do. Imagine someone who would add support for further filters. He would then only need to change a single file, instead of correctly modifying multiple distinct places within several files that also contain other non related things.